src/policies.rs

```diff
@@ -355,6 +368,73 @@ fn verify_different_authors<G: Git>(
     }
 }
 
+fn verify_rebased<G: Git>(
+    commits: &[Commit],
+    git: &G,
+    old_commit_id: Oid,
+    new_commit_id: Oid,
+    ref_name: &str,
+    override_tag_pattern: &Option<String>,
+) -> Result<PolicyResult, Box<dyn Error>> {
+    let new_branch = old_commit_id.is_zero();
+    let is_merge = git.is_merge_commit(new_commit_id);
+    let is_mainline = git.is_mainline(ref_name)?;
+    let new_commit = git.find_commit(new_commit_id, override_tag_pattern)?;
+    let is_not_fast_forward = !git.is_descendent_of(new_commit_id, old_commit_id)?;
+
+    if !is_mainline {
+        info!(
+            "Rebase verification passed for {}: Not updating a mainline branch",
+            new_commit_id
+        );
+        Ok(PolicyResult::Ok)
+    } else if !is_merge {
+        info!(
+            "Rebase verification passed for {}: Not a merge commit",
+            new_commit_id
+        );
+        Ok(PolicyResult::Ok)
+    } else if new_branch {
+        info!("Rebase verification passed for {}: New branch does not require being rebased for a merge commit", new_commit_id);
+        Ok(PolicyResult::Ok)
+    } else if commits.len() == 0 {
+        info!(
+            "Rebase verification passed for {}: No new commits pushed",
+            new_commit_id
+        );
+        Ok(PolicyResult::Ok)
+    } else if is_not_fast_forward {
+        info!(
+            "Rebase verification passed for {0}: Commit Id {0} is not a descendent of Commit Id {1}, it is most likely that a force-push has occurred",
+            new_commit_id,
+            old_commit_id
+        );
+        Ok(PolicyResult::Ok)
+    } else {
+        let new_commit_is_rebased = new_commit
+            .parents
+            .iter()
+            .map(|parent_id| {
+                git.is_descendent_of(*parent_id, old_commit_id)
+                    .map(|is_descendent| is_descendent || *parent_id == old_commit_id)
```

Comment:
When is the `parent_id` the `old_commit_id`?

Reply:
This condition covers the situation when there is an "octopus merge" where one of the streams was just the destination node and the old_commid_id node - from what I understand anyway.

Reply:
There's maybe some confusion here. We need to iterate over all parents and check that they're all good, rather than just finding the one branch being merged in because of the possibility of an octopus merge.

The parent_id == old_commit_id case is because in the default case where you've clicked merge in github one of your merge commit's parents is master's old commit_id.